server/doc.go

```diff
@@ -0,0 +1,34 @@
+// Package server contains every kpng's function related to serving its API.
+//
+// The general approach is that you want to provide a [proxystore.Store] and make something of it.
+// This store contains what we call the global model, representing the cluster-wide information
+// needed by kpng to provide what we call the local model, representing the node-level information.
+//
+// There are 3 ways to provide a [proxystore.Store], all using jobs:
+//   - [jobs/kube2store]: from a k8s' apiserver;
+//   - [jobs/api2store]: from another kpng instance serving the global API;
+//   - [jobs/file2store]: from a serialized global model.
+//
+// From a [proxystore.Store], you can:
+//   - serve it over the kpng's API ([jobs/store2api]);
+//   - serialize the global model to a file ([jobs/store2file]);
+//   - stream the changes to the global model ([jobs/store2globaldiff]);
+//   - stream the changes to the local model ([jobs/store2localdiff]).
```

Comment:
Maybe we should add a note on the `watchstate` package here? It's pretty crucial for the streaming cases right? 

Something like:

The WatchState Package is what we use internally to actually facility the streaming of Diffs to the backend Sinks?

Reply:
I didn't go inside pkg in this iteration. We can add more later, esp. you as you're going through it :)